torch/testing/_internal/logging_utils.py

```diff
@@ -0,0 +1,77 @@
+import torch._dynamo.test_case
+import unittest.mock
+import os
+import contextlib
+
+import logging
+
+def log_settings(settings):
+    return unittest.mock.patch.dict(os.environ, {"TORCH_LOGS": settings})
+
+def make_test(settings, log_names, install_hook=None):
+    def wrapper(fn):
+        def test_fn(self):
+            records = []
+            with log_settings(settings), self._handler_watcher([logging.getLogger(n) for n in log_names], records, install_hook):
+                fn(self, records)
+
+        return test_fn
+
+    return wrapper
+
+
+class LoggingTestCase(torch._dynamo.test_case.TestCase):
```

Comment:
Want comment describing high level testing strategy

Reply:
This has been very simplified now, but still added the comment